app/src/org/commcare/activities/GlobalPrivilegeClaimingActivity.java

```diff
@@ -0,0 +1,224 @@
+import android.view.View;
+import android.widget.Button;
+import android.widget.TextView;
+import android.widget.Toast;
+
+import org.commcare.dalvik.R;
+import org.commcare.preferences.CommCarePreferences;
+import org.commcare.preferences.GlobalPrivilegesManager;
+import org.commcare.utils.SigningUtil;
+import org.commcare.utils.StringUtils;
+import org.javarosa.core.services.locale.Localization;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+
+/**
+ * Activity that allows a user to scan a barcode from HQ to enable a global privilege, such as
+ * superuser mode
+ *
+ * @author Aliza Stone (astone@dimagi.com), created 6/9/16.
+ */
+public class GlobalPrivilegeClaimingActivity extends Activity {
+
+    private static final String TAG = GlobalPrivilegeClaimingActivity.class.getSimpleName();
+    public static final String KEY_PRIVILEGE_NAME = "key-privilege-name";
+
+    // activity request codes
+    public static final int BARCODE_CAPTURE = 1;
+
+    // menu item IDs
+    private static final int DISABLE = 1;
+
+    // the name of the privilege that this activity is set up to claim/revoke
+    private String privilegeName;
+
+    private String privilegeDisplayName;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
+        privilegeName = getIntent().getStringExtra(KEY_PRIVILEGE_NAME);
+        if (!GlobalPrivilegesManager.allGlobalPrivilegesList.contains(privilegeName)) {
+            setResult(RESULT_CANCELED);
+            finish();
+        }
+        privilegeDisplayName = GlobalPrivilegesManager.getPrivilegeDisplayName(privilegeName);
+
+        setContentView(R.layout.privilege_claiming_view);
+        findViewById(R.id.claim_button).setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                callOutToBarcodeScanner();
+            }
+        });
+        ((TextView)findViewById(R.id.instructions)).setText(GlobalPrivilegesManager.getInstructionsTextId(this.privilegeName));
+
+        CommCarePreferences.addBackButtonToActionBar(this);
+    }
+
+    @Override
+    protected void onResume() {
+        super.onResume();
+        refreshUI();
+    }
+
+    private void refreshUI() {
+        TextView enabledTextView = (TextView)findViewById(R.id.enabled_textview);
+        TextView notEnabledTextView = (TextView)findViewById(R.id.not_enabled_textview);
+        Button claimButton = (Button)findViewById(R.id.claim_button);
+        TextView instructions = (TextView)findViewById(R.id.instructions);
+
+        if (GlobalPrivilegesManager.isPrivilegeEnabled(this.privilegeName)) {
+            notEnabledTextView.setVisibility(View.GONE);
+            claimButton.setVisibility(View.GONE);
+            instructions.setVisibility(View.GONE);
+            enabledTextView.setVisibility(View.VISIBLE);
+            enabledTextView.setText(getEnabledText());
```

Comment:
Can you lift this block into a helper that mentions the state the UI is currently in? Like `privilegeEnabledUIState`